clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -5591,6 +5614,20 @@ bool SYCLIntegrationFooter::emit(raw_ostream &OS) {
       DeviceGlobOS << SYCLUniqueStableIdExpr::ComputeName(S.getASTContext(),
                                                           VD);
       DeviceGlobOS << "\");\n";
+    } else if (isSyclType(VD->getType(), SYCLTypeAttr::host_pipe)) {
```

Comment:
There is quite a lot of repetition between `device_global` and `host_pipe` code. Can we generalize it?

Reply:
I see what you're saying, although I thought it would make it a little messy to e.g., use trinary operators to control the class name output. And I think the device global and host pipe class names need to be separate here because the RT needs to treat these differently.